code/datums/components/growth_and_differentiation.dm

```diff
@@ -56,6 +70,8 @@
 
 /datum/component/growth_and_differentiation/UnregisterFromParent()
 	UnregisterSignal(SSticker, COMSIG_TICKER_ROUND_STARTING)
+	for(var/signal in signals_to_kill_on)
+		UnregisterSignal(parent, signal)
```

Comment:
why do we implement this in the first place? will this component be moved around at any point?

Reply:
Also you can just pass a list into unreg signal and it'll work, it's even slightly faster

Reply:
no because we qdel ourselves when growth is complete. i think admin schenanigans are the only real reason.

Reply:
you don't need to support admin component transferring if you don't want to, I'm not sure we do in the first place

Reply:
it doesn't seem like the hardest thing to support